### PR TITLE
Try to kill without descriptors leak

### PR DESCRIPTION
**Pull request by KVS85**
https://arangodb.atlassian.net/jira/software/projects/COP/boards/8?selectedIssue=COP-152

### CODE REVIEW

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/view/Windows/job/arangodb-matrix-pr-windows/443/

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/557/
http://jenkins.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/558/
http://jenkins.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/559/

File: scripts/lib/Utils.psm1 [hunk 1]
```diff
@@ -184,7 +184,7 @@ Function launchTest($which) {
     Set-Location $global:ARANGODIR; comm
     $arangosh = "$global:ARANGODIR\build\bin\$BUILDMODE\arangosh.exe"
     $test = $global:launcheableTests[$which]
-    Write-Host "Test: " $test['testname'] " - " $test['identifier']
+    Write-Host "Test: " $test['testname'] " - " $test['identIfier']
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert this change.

File: scripts/lib/Utils.psm1 [hunk 2]
```diff
@@ -207,7 +207,7 @@ Function launchTest($which) {
     Pop-Location
 }
 
-Function registerTest($testname, $index, $bucket, $filter, $moreParams, $cluster, $weight, $sniff)
+Function registerTest($testname, $index, $bucket, $filter, $moreParams, $cluster, $weight, $snIff)
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert this change.

File: scripts/lib/Utils.psm1 [hunk 3]
```diff
@@ -237,16 +237,16 @@ Function registerTest($testname, $index, $bucket, $filter, $moreParams, $cluster
...
             $dumpAgencyOnError = "true"
         }
-        if ($weight) {
+        If ($weight) {
           $testWeight = $weight
         }
 
-        if ($sniff) {
-          $testparams = $testparams + " --sniff true --sniffProgram `"$global:TSHARK`" --sniffDevice $global:dumpDevice"
+        If ($snIff) {
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert changing the variable name - attribute name change below too.

File: scripts/lib/Utils.psm1 [hunk 4]
```diff
@@ -265,7 +265,7 @@ Function registerTest($testname, $index, $bucket, $filter, $moreParams, $cluster
           running=$false;
           weight=$testWeight;
         testname=$testname;
-        identifier=$output;
+        identIfier=$output;
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert this change

File: scripts/lib/Utils.psm1 [hunk 5]
```diff
@@ -292,7 +306,7 @@ Function LaunchController($seconds)
     $numberTestsSlots = [math]::Round($global:numberSlots * 0.9) # Should leave 10% of slots free for $global:numberSlots > 4
     While (($seconds -gt 0) -and (($currentRunning -gt 0) -or ($nextLauncheableTest -lt $maxLauncheableTests))) {
         while (($currentScore -lt $numberTestsSlots) -and ($nextLauncheableTest -lt $global:maxTestCount)) {
-            Write-Host "Launching $nextLauncheableTest '" $global:launcheableTests[$nextLauncheableTest ]['identifier'] "'"
+            Write-Host "Launching $nextLauncheableTest '" $global:launcheableTests[$nextLauncheableTest ]['identIfier'] "'"
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert this change.

File: scripts/lib/Utils.psm1 [hunk 6]
```diff
@@ -282,6 +282,20 @@ Function registerTest($testname, $index, $bucket, $filter, $moreParams, $cluster
...
 
+Function Kill-Children ($Pid, $SessionId)
+{
+    Get-WmiObject win32_process | Where {$_.ParentProcessId -eq $Pid -And $_.SessionId -eq $SessionId -And -Not [string]::IsNullOrEmpty($_.Path) }) | ForEach-Object { Kill-Children $_.ProcessId $_.SessionId }
+    If (Get-Process -Id $Pid -ErrorAction SilentlyContinue)
+    {
+        Write-Host "Killing child: $Pid"
+
+        # Try to avoid https://wiki.jenkins.io/display/JENKINS/Spawning+processes+from+build:
+        handle.exe -p $Pid | Where {$_ -match "^\s*([0-9A-F]+): File..*$" } | ForEach { $h = $Matches[1]; handle.exe -c $h -p $Pid -y | Out-Null}
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: there really is no powershell internal version of `handle.exe`?

File: scripts/lib/Utils.psm1 [hunk 7]
```diff
@@ -302,15 +316,15 @@ Function LaunchController($seconds)
         $currentRunning = 0
         $currentRunningNames = @()
         ForEach ($test in $global:launcheableTests) {
-            if ($test['running']) {
-                if ($test['process'].HasExited) {
+            If ($test['running']) {
+                If ($test['process'].HasExited) {
                     $currentScore = $currentScore - $test['weight']
-                    Write-Host "$((Get-Date).ToUniversalTime().ToString('yyyy-MM-ddTHH.mm.ssZ')) Testrun finished: "$test['identifier'] $test['launchdate']
+                    Write-Host "$((Get-Date).ToUniversalTime().ToString('yyyy-MM-ddTHH.mm.ssZ')) Testrun finished: "$test['identIfier'] $test['launchdate']
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert this change.

File: scripts/lib/Utils.psm1 [hunk 8]
```diff
@@ -302,15 +316,15 @@ Function LaunchController($seconds)
...
                     $str=$($test | where {($_.Name -ne "commandline")} | Out-String)
                     $test['running'] = $false
                 }
                 Else {
-                    $currentRunningNames += $test['identifier']
+                    $currentRunningNames += $test['identIfier']
```
Comment (dothebart) [inline scripts/lib/Utils.psm1]: please revert this change.

File: scripts/lib/Utils.psm1 (hunk 6 above)
Comment (KVS85) [reply scripts/lib/Utils.psm1]: There is something custom: https://www.powershellgallery.com/packages/PoshInternals/1.0.34/Content/Handle.ps1
